core/protocols/aws-query-protocol/src/test/java/software/amazon/awssdk/protocols/query/interceptor/QueryParametersToBodyInterceptorTest.java

```diff
@@ -77,35 +96,57 @@ public void postWithContentIsUnaltered() throws Exception {
 
     @Test
     public void onlyAlterRequestsIfParamsArePresent() throws Exception {
-        SdkHttpFullRequest.Builder mutableRequest = ValidSdkObjects.sdkHttpFullRequest()
-                                                                   .clearHeaders()
-                                                                   .method(SdkHttpMethod.POST);
+        SdkHttpFullRequest request = requestBuilder.clearQueryParameters().build();
 
-        SdkHttpFullRequest output = sut.execute(mutableRequest, requestContext(mutableRequest)).build();
+        SdkHttpFullRequest output = (SdkHttpFullRequest) interceptor.modifyHttpRequest(new Context(request, null),
+                                                                                       executionAttributes);
 
         assertThat(output.rawQueryParameters()).hasSize(0);
         assertThat(output.headers()).hasSize(0);
         assertThat(output.contentStreamProvider()).isEmpty();
     }
 
     private void nonPostRequestsUnaltered(SdkHttpMethod method) {
-        SdkHttpFullRequest.Builder mutableRequest = ValidSdkObjects.sdkHttpFullRequest()
-                                                                   .method(method)
-                                                                   .clearHeaders()
-                                                                   .putRawQueryParameter("key", singletonList("value"));
 
-        SdkHttpFullRequest output = sut.execute(mutableRequest, requestContext(mutableRequest)).build();
+        SdkHttpFullRequest request = requestBuilder.method(method).build();
+
+        SdkHttpFullRequest output = (SdkHttpFullRequest) interceptor.modifyHttpRequest(new Context(request, null),
+                                                                                       executionAttributes);
 
         assertThat(output.rawQueryParameters()).hasSize(1);
         assertThat(output.headers()).hasSize(0);
         assertThat(output.contentStreamProvider()).isEmpty();
     }
 
-    private RequestExecutionContext requestContext(SdkHttpFullRequest.Builder mutableRequest) {
-        ExecutionContext executionContext = ClientExecutionAndRequestTimerTestUtils.executionContext(mutableRequest.build());
-        return RequestExecutionContext.builder()
-                                      .executionContext(executionContext)
-                                      .originalRequest(NoopTestRequest.builder().build())
-                                      .build();
+    public final class Context implements software.amazon.awssdk.core.interceptor.Context.ModifyHttpRequest {
```

Comment:
If we do have to do this maybe a slightly less generic name that gives the reader a clue as to why we had to do this (eg: ContextWithNoRequest or whatever)